### DISCUSSION TITLE
LC for AL GO

### DISCUSSION THREAD

**Discussion by cegekaJG** [StefanMaron/BusinessCentral.LinterCop#519] [Q&A]
Is there a way to download the LinterCop DLL without editing the default workflow? I'd like to keep the default as much intact as possible for the sake of compatibility, but since we've started customizing LC rule actions for our projects, I'd like our CI/CD pipelines to do the same.
I understand that [this blog post](https://robvanbekkum.nl/additional-build-steps/#custom-code-analyzer-in-build-pipelines) is already a good explanation on how to retrieve a build, but this requires editing the workflow YAML.
Is it possible to include LC to AL Go's CI/CD using the configuration files only?

**Comment by Arthurvdv** [comment]
I'm not (yet) familiar with the possibilities of AL-Go, where I mostly work in Azure DevOps myself.

The documentation states that you can pass a URL for a CustomCodeCops it seems.
![image](https://github.com/StefanMaron/BusinessCentral.LinterCop/assets/44637996/7368b0a2-c0bc-46cb-a0c4-6434bac3879f)
https://github.com/microsoft/AL-Go/blob/main/Scenarios/settings.md#advanced-settings

Can you try to pass the this URL and see if this works?
`https://github.com/StefanMaron/BusinessCentral.LinterCop/releases/latest/download/BusinessCentral.LinterCop.dll`

**Reply by cegekaJG** [reply]
That URL is exactly what I was looking for, I wasn't sure if there was an unzipped link to the file.
It should work, but it didn't when I tried it - though I think that's on AL Go. There was a [ticket](https://github.com/microsoft/AL-Go/issues/788) about it back in October - which I've commented on - so I hope I can get some info.

**Reply by dsaveyn** [reply]
I can report this customCodeCops property is still not working (tested it myself yesterday).

**Reply by cegekaJG** [reply]
IMPORTANT: As of `v3.60.0`, `https://github.com/StefanMaron/BusinessCentral.LinterCop/releases/latest/download/BusinessCentral.LinterCop.current.dll` is no longer a valid URL. Use `https://github.com/StefanMaron/BusinessCentral.LinterCop/releases/latest/download/BusinessCentral.LinterCop.dll` instead.

**Reply by Arthurvdv** [reply]
Thanks for posting this here. I’ve edited the url on the previous post and removed the .current.

More details on this: [Artifacts .current.dll and .next.dll are obsolete](https://github.com/StefanMaron/BusinessCentral.LinterCop/issues/709)

**Comment by freddydk** [comment]
The reason for this not working is, that despite the documentation saying so, CustomCodeCops cannot be a URL today in AL-Go for GitHub.
The fix for this is in BcContainerHelper and in a PR here: https://github.com/microsoft/navcontainerhelper/pull/3488 - which means that BcContainerHelper also will support specifying a URL as customCodeCop for just-in-time download.

When that has merged, you would need to add "vsixFile": "latest" to your settings file as well as the Lintercop from @StefanMaron requires the latest AL Language Exrension.

You can see a test run with the Lintercop here: https://github.com/BusinessCentralApps/buildorder/actions/runs/8679462223

**Comment by cegekaJG** [comment]
Do I need to wait for a new release or should it work immediately?

**Reply by Arthurvdv** [reply]
This will be avaiable on the version [6.0.16](https://github.com/microsoft/navcontainerhelper/blob/f157a40f342e08a9e8b3534de933b84a093ae83b/ReleaseNotes.txt#L4) of the BcContainerHelper.

![image](https://github.com/StefanMaron/BusinessCentral.LinterCop/assets/44637996/050b7444-8cb1-4d76-8c42-31c592d7fe21)
https://www.powershellgallery.com/packages/BcContainerHelper

Currently 6.0.15 is de current version, so we need to wait for the release of 6.0.16.

**Comment by Arthurvdv** [comment]
Awesome news, the version 6.0.16 of the BcContainerHelper is released and it works perfectly 🤗

![image](https://github.com/StefanMaron/BusinessCentral.LinterCop/assets/44637996/819793ec-10bd-4675-92d3-bc64d283f68c)

In the settings.json file you can specify the url to dll file of the LinterCop (and enable downloading the latest VSIX).

```json
{
  "vsixFile": "latest",
  "customCodeCops": [
    "https://github.com/StefanMaron/BusinessCentral.LinterCop/releases/latest/download/BusinessCentral.LinterCop.dll"
  ]
}
```

I'll update the documentation on using the LinterCop with AL-Go in the next few days.
